enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    my_each do |element|
+      selected_array << element if yield(element)
+    end
+    selected_array
+  end
+
+  def my_all?(_para = nil)
+    result = true
+    if block_given?
+      my_each do |element|
+        result = false unless yield(element)
+      end
+    else
+      my_each do |element|
+        result = false unless element
+      end
+    end
+    result
+  end
+
+  def my_any?(para = nil)
+    return my_any?(para) if block_given? && !para.nil?
+
+    if block_given?
+      to_a.my_each do |element|
+        return true if yield element
+
+        false
+      end
+    elsif para
+      to_a.my_each do |element|
+        return true if element == para
+      end
+    elsif para.nil?
+      to_a.my_each do |element|
+        return true if element
+      end
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving the wrong answer
![my_any1](https://user-images.githubusercontent.com/52209067/80103735-9bfb0600-8532-11ea-9841-6d30fe161ff6.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/o/) == %w[dog door rod blade].any?(/o/) => true
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80103826-c947b400-8532-11ea-9b0b-0450ad7b7e8b.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/80103870-de244780-8532-11ea-97b6-94f7568614de.png)